src/accountsdb/db.zig

```diff
@@ -1855,37 +1877,52 @@ pub const AccountsDB = struct {
     pub fn getAccountFromRef(self: *Self, account_ref: *const AccountRef) !Account {
         switch (account_ref.location) {
             .File => |ref_info| {
-                return try self.getAccountInFile(
-                    self.allocator,
-                    ref_info.file_id,
-                    ref_info.offset,
-                );
+                const account_in_cache = blk: {
+                    const accounts_cache, var accounts_cache_lg = self.accounts_cache.writeWithLock();
```

Comment:
read lock?

Reply:
The .get requires a mutable pointer, as it can modify the lru state (previously the lru state was behind a pointer).